src/rpdk/core/project.py

```diff
@@ -338,6 +339,10 @@ def validate_and_load_resource_settings(self, raw_settings):
         self._plugin = load_plugin(raw_settings["language"])
         self.settings = raw_settings.get("settings", {})
         self.canary_settings = raw_settings.get("canarySettings", {})
+        if raw_settings.get("canarySettings", False) is False:
```

Comment:
A separate property was added here to avoid having `canary_settings` being either an object or boolean.